pkg/controllers/disruption/consolidation_test.go

```diff
@@ -134,6 +134,54 @@ var _ = Describe("Consolidation", func() {
 			Expect(recorder.Calls("Unconsolidatable")).To(Equal(6))
 		})
 	})
+	Context("Metrics", func() {
+		const eligibleNodesMetric = "karpenter_disruption_eligible_nodes"
```

Comment:
Totally, don't know why I didn't 🤦. AFAIK there shouldn't be any cyclic dep considering this is the test suite.

Reply:
Got around to rechecking, thought there was but there aren't any constants here. Could go either way on making some or just defining one here. Leaning towards just keeping it as is, this is what we do elsewhere in the project.

Reply:
WDYT about making this more programmatic so that if we update the metric name in the future the test automatically updates? Also makes helps to not mistyping metrics, and get compiler checks. Maybe we could make a helper that gets the namespace, subsystem, and name of the metric and joins it into the metric name?

Reply:
So I was able to figure it out but it is...interesting to say the least. Doesn't seem like there's anyway to just get the qualified name, only the full description which you need to extract the name from. Also, need to use an async API to get this description from all collector types (looking at gauge vecs). Seems to work fine now that it's implemented though 🤷 .